src/main/java/ch/njol/skript/expressions/ExprPickles.java

```diff
@@ -0,0 +1,151 @@
+	"\tsend \"Wow! This stack of sea pickles contained %event-block's pickles% pickles!\"",
+	"\tsend \"It could've held a maximum of %event-block's maximum pickles% pickles!\"",
+	"\tsend \"It had to have held at least %event-block's minimum pickles% pickles!\"",
+	"\tcancel event",
+	"\tset event-block's pickles to event-block's maximum pickles",
+	"\tsend \"This bad boy is going to hold so many pickles now!!\""
+})
+@Since("INSERT VERSION")
+public class ExprPickles extends SimplePropertyExpression<Block, Long> {
+
+	static {
+		register(ExprPickles.class, Long.class, "[:minimum|:maximum] pickles", "block");
+	}
+
+	private boolean minimum, maximum;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		minimum = parseResult.hasTag("minimum");
+		maximum = parseResult.hasTag("maximum");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Long convert(Block block) {
+		BlockData blockData = block.getBlockData();
+		if (!(blockData instanceof SeaPickle))
+			return null;
+
+		SeaPickle pickleData = (SeaPickle) blockData;
+
+		if (maximum)
+			return (long) pickleData.getMaximumPickles();
+		if (minimum)
+			return (long) pickleData.getMinimumPickles();
+		return (long) pickleData.getPickles();
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (minimum || maximum)
+			return null;
+		switch (mode) {
+			case SET:
+			case ADD:
+			case REMOVE:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(Number.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (delta == null && mode != ChangeMode.RESET && mode != ChangeMode.DELETE)
+			return;
+
+		int change = delta != null ? ((Number) delta[0]).intValue() : 0;
+		if (mode == ChangeMode.REMOVE)
+			change *= -1;
+
+		for (Block block : getExpr().getArray(event)) {
+
+			// Obtain pickle data
+			BlockData blockData = block.getBlockData();
+			if (!(blockData instanceof SeaPickle))
+				return;
+			SeaPickle pickleData = (SeaPickle) blockData;
+
+			int newPickles = change;
+
+			// Calculate new pickles value
+			switch (mode) {
+				case ADD:
+				case REMOVE:
+					newPickles += pickleData.getPickles();
+				case SET:
+					newPickles = Math.max(pickleData.getMinimumPickles(), newPickles); // Ensure value isn't too low
+					newPickles = Math.min(pickleData.getMaximumPickles(), newPickles); // Ensure value isn't too high
+					break;
+				case RESET:
+				case DELETE:
+					newPickles = pickleData.getMinimumPickles();
+					break;
+				default:
+					assert false;
+			}
+
+			// Update the block data
+			pickleData.setPickles(newPickles);
+			block.setBlockData(pickleData);
+
+		}
+	}
+
+	@Override
+	public Class<? extends Long> getReturnType() {
+		return Long.class;
+	}
+
+	@Override
+	protected String getPropertyName() {
```

Comment:
> I think this should be `public` tho in superclass it isn't which I understand why but it may be even better to change that (in another PR) because some docs may use it?

Why should it be public? That method is only used once inside the superclass and it's used for the toString method. The only reason it's not private is because it needs to be implemented by the subclasses

Reply:
my thoughts is that it may be used by some docs, not idea how yet but maybe therefore it it's optional to make it public